yahoo_panoptes/framework/resources.py

```diff
@@ -986,10 +984,13 @@ def close_resource_cache(self):
         Returns:
             None
         """
-        self._db.close()
+        if self._db:
+            self._db.close()
+        else:
+            raise PanoptesResourceError("Attempted to close connection to SQLite db that was not open")
```

Comment:
I have mixed feelings about raising this exception vs. just logging the exception.

Reply:
I think just logging the exception is the right way to go. Presumably if close_resource_cache is being called, the user has already accomplished what they wanted with it and the in-memory db will be wiped anyway as that process exits.